src/sgame/sg_admin.cpp

```diff
@@ -4255,7 +4255,15 @@ bool G_admin_restart( gentity_t *ent )
 	}
 
 	// check that the layout's available
-	builtin = !*layout || !Q_stricmp( layout, S_BUILTIN_LAYOUT );
+	builtin = !Q_stricmp( layout, S_BUILTIN_LAYOUT );
+
+	if( !*layout )
+	{
+		strcpy( layout, "default" );
+		if ( !trap_FS_FOpenFile( va( "layouts/%s/%s.dat", map, layout ), nullptr, fsMode_t::FS_READ ) ) {
```

Comment:
I think this return value check is wrong. It returns -1 if the file doesn't exist.

Reply:
this if statement should be true when no file exists

Reply:
this if statement should be true when no file exists

Reply:
It's only true if the file exists but has a length of zero.

Reply:
hm i think i get what you mean. i tried the code and strangely this works as intended 🤷‍♂️
but i just realized that this check is not necessary at this place